Subsurface/Node/Common/peripherals/inc/I2CInterface.h

```diff
@@ -0,0 +1,56 @@
+#ifndef NODE_I2CINTERFACE_H
+#define NODE_I2CINTERFACE_H
+
+#include "main.h"
+
+class I2CInterface
+{
+    private:
+
+    // Pointer to I2C Handle Type Def structure contains the config info for the specified I2C.
+    I2C_HandleTypeDef *m_hi2c;
+    uint8_t m_dataReceived[8];
+    uint8_t m_bufferLength = 8;
+    
+    public:
+
+    // Constructor
+    I2CInterface(I2C_HandleTypeDef *hi2c);
+
+    // Returns a pointer to the data received buffer
+    uint8_t* getReceivedData() {  return m_dataReceived; }
+
+    // Get state, and error methods
+    HAL_I2C_StateTypeDef getState() { return HAL_I2C_GetState(m_hi2c); }
+    uint32_t getError() { return HAL_I2C_GetError(m_hi2c); }
+
+    // Getter and setter for I2C addressing mode (10 bit or 8 bit)
+    void setAddressingMode(uint32_t mode) { (*m_hi2c).Init.AddressingMode = mode; }
+    uint32_t getAddressingMode() { return (*m_hi2c).Init.AddressingMode; }
+
+    // Master transmit and receive functions (polling mode)
+    HAL_StatusTypeDef masterTransmit(uint16_t address, uint8_t* data, uint32_t timeout, uint8_t size);   
+    HAL_StatusTypeDef masterReceive(uint16_t address, uint32_t timeout);
+
+    // Reading / writing from device internal memory addresses 
+    HAL_StatusTypeDef memWrite(uint8_t* data, uint16_t devAddress, uint16_t memAddress, 
+                                uint16_t memAddressSize, uint8_t size, uint32_t timeout);
```

Comment:
I think this formatting looks fine, but for future reference i like when you have one parameter per line if it is going to be too long to all fit on one line. For example:
```C
    HAL_StatusTypeDef memWrite(uint8_t* data,
                               uint16_t devAddress,
                               uint16_t memAddress, 
                               uint16_t memAddressSize,
                               uint8_t size,
                               uint32_t timeout);
```
Only thing I will say is that it would be nice if the first parameter on the second line lined up with the first parameter on the first line, but this is nit-picky

Reply:
noted, will make sure to fix and continue this format moving forward